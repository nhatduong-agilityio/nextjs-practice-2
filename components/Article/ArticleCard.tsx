// Libs
import React from 'react'
import {
  Card as ChakraCard,
  CardBody,
  CardHeader,
  Heading,
  Text,
  VStack,
  HStack,
} from '@chakra-ui/react'
import type { StaticImageData } from 'next/legacy/image'
import NextLink from 'next/link'

// Components
import ArrowForwardIcon from '@components/Icons/ArrowForwardIcon'
import Image from '../Common/Image'

type TArticle = {
  id: string
  title: string
  image: StaticImageData
  imageAlt: string
  description: string
}

interface IProps {
  item: TArticle
  pathLink?: string
}

const ArticleCard = ({ item, pathLink }: IProps) => (
  <ChakraCard boxShadow='none'>
    <CardHeader p={0} flexDir='row'>
      <Image
        alt={item.imageAlt}
        src={item.image}
        variant='primary'
        width='full'
        height={['150px', '250px']}
      />
    </CardHeader>
    <CardBody p={0} pt='15px'>
      <VStack style={{ margin: 0 }} w='full' alignItems='flex-start' gap={0}>
        <Heading variant={['boldSM', 'boldMD']} lineHeight={0}>
          {item.title}
        </Heading>
        <Text size={['16', '18']} noOfLines={2} variant='grayScheme' pt={['10px', '15px']}>
          {item.description}
        </Text>
        {pathLink && (
          <NextLink href={pathLink} passHref>
            <HStack flex='row' pt='15px' fontSize={['16px', '18px']}>
              <Text size={['16', '18']} noOfLines={1} maxW='80%' variant='small'>
                Read more about {item.title}
              </Text>
              <ArrowForwardIcon
                width={['16px', '20px']}
                height={['16px', '20px']}
                viewBox='0 0 14 10'
                fill='black'
                ml='10px'
              />
            </HStack>
          </NextLink>
        )}
      </VStack>
    </CardBody>
  </ChakraCard>
)

export default ArticleCard
