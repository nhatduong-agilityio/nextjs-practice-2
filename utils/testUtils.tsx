import { ChakraProvider } from '@chakra-ui/react'
import { AuthProvider } from '@store/providers/AuthProvider'
import { LoadingProvider } from '@store/providers/LoadingProvider'
import type { RenderOptions, RenderResult } from '@testing-library/react'
import { render } from '@testing-library/react'
import { customTheme } from '@themes/index'
import type { ReactElement } from 'react'

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return (
    <ChakraProvider theme={customTheme}>
      <LoadingProvider>
        <AuthProvider>{children}</AuthProvider>
      </LoadingProvider>
    </ChakraProvider>
  )
}

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>): RenderResult =>
  render(ui, { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
