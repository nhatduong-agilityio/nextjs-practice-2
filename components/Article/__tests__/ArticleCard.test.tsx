// Utils
import { render } from '@utils/testUtils'

import { ArticlesMock } from '@mock/dataMock'
import type { TArticle } from '@matched-types/articles'

// Components
import ArticleCard from '../ArticleCard'

describe('ArticleCard render', () => {
  const propsMock = {
    item: ArticlesMock[0] ? ArticlesMock[0] : ({} as TArticle),
  }

  it('Should ArticleCard match Snapshot', () => {
    const { container } = render(<ArticleCard {...propsMock} />)

    expect(container).toMatchSnapshot()
  })
})
