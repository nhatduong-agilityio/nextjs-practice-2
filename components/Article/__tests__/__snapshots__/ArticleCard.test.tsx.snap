// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleCard render Should ArticleCard match Snapshot 1`] = `
<div>
  <div
    class="chakra-card css-1fnaag1"
  >
    <div
      class="chakra-card__header css-e3ulnx"
    >
      <div
        class="css-z6z70s"
      >
        <span
          style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
        >
          <img
            alt="Picture of furniture chair and table office"
            data-nimg="fill"
            decoding="async"
            sizes="(max-width: 1440px) 100vw, (max-width: 1536px) 50vw, 33vw"
            src="/_next/image?url=https%3A%2F%2Ffirebasestorage.googleapis.com%2Fv0%2Fb%2Fshop-bag-ffd59.appspot.com%2Fo%2Fimages%252Farticle-1.jpg%3Falt%3Dmedia%26token%3Dc1a5e572-6861-4b4d-86da-b047368f8a2c&w=3840&q=75"
            srcset="/_next/image?url=https%3A%2F%2Ffirebasestorage.googleapis.com%2Fv0%2Fb%2Fshop-bag-ffd59.appspot.com%2Fo%2Fimages%252Farticle-1.jpg%3Falt%3Dmedia%26token%3Dc1a5e572-6861-4b4d-86da-b047368f8a2c&w=256&q=75 256w, /_next/image?url=https%3A%2F%2Ffirebasestorage.googleapis.com%2Fv0%2Fb%2Fshop-bag-ffd59.appspot.com%2Fo%2Fimages%252Farticle-1.jpg%3Falt%3Dmedia%26token%3Dc1a5e572-6861-4b4d-86da-b047368f8a2c&w=384&q=75 384w, /_next/image?url=https%3A%2F%2Ffirebasestorage.googleapis.com%2Fv0%2Fb%2Fshop-bag-ffd59.appspot.com%2Fo%2Fimages%252Farticle-1.jpg%3Falt%3Dmedia%26token%3Dc1a5e572-6861-4b4d-86da-b047368f8a2c&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Ffirebasestorage.googleapis.com%2Fv0%2Fb%2Fshop-bag-ffd59.appspot.com%2Fo%2Fimages%252Farticle-1.jpg%3Falt%3Dmedia%26token%3Dc1a5e572-6861-4b4d-86da-b047368f8a2c&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Ffirebasestorage.googleapis.com%2Fv0%2Fb%2Fshop-bag-ffd59.appspot.com%2Fo%2Fimages%252Farticle-1.jpg%3Falt%3Dmedia%26token%3Dc1a5e572-6861-4b4d-86da-b047368f8a2c&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Ffirebasestorage.googleapis.com%2Fv0%2Fb%2Fshop-bag-ffd59.appspot.com%2Fo%2Fimages%252Farticle-1.jpg%3Falt%3Dmedia%26token%3Dc1a5e572-6861-4b4d-86da-b047368f8a2c&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Ffirebasestorage.googleapis.com%2Fv0%2Fb%2Fshop-bag-ffd59.appspot.com%2Fo%2Fimages%252Farticle-1.jpg%3Falt%3Dmedia%26token%3Dc1a5e572-6861-4b4d-86da-b047368f8a2c&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Ffirebasestorage.googleapis.com%2Fv0%2Fb%2Fshop-bag-ffd59.appspot.com%2Fo%2Fimages%252Farticle-1.jpg%3Falt%3Dmedia%26token%3Dc1a5e572-6861-4b4d-86da-b047368f8a2c&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Ffirebasestorage.googleapis.com%2Fv0%2Fb%2Fshop-bag-ffd59.appspot.com%2Fo%2Fimages%252Farticle-1.jpg%3Falt%3Dmedia%26token%3Dc1a5e572-6861-4b4d-86da-b047368f8a2c&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Ffirebasestorage.googleapis.com%2Fv0%2Fb%2Fshop-bag-ffd59.appspot.com%2Fo%2Fimages%252Farticle-1.jpg%3Falt%3Dmedia%26token%3Dc1a5e572-6861-4b4d-86da-b047368f8a2c&w=3840&q=75 3840w"
            style="width: 0px; height: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover; background-size: cover; background-position: 0% 0%; filter: blur(20px); background-image: url(data:image/svg+xml;base64,Cjxzdmcgd2lkdGg9IjQwMCIgaGVpZ2h0PSIzMDAiIHZlcnNpb249IjEuMSIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB4bWxuczp4bGluaz0iaHR0cDovL3d3dy53My5vcmcvMTk5OS94bGluayI+CiAgPGRlZnM+CiAgICA8bGluZWFyR3JhZGllbnQgaWQ9ImciPgogICAgICA8c3RvcCBzdG9wLWNvbG9yPSIjY2NjIiBvZmZzZXQ9IjIwJSIgLz4KICAgICAgPHN0b3Agc3RvcC1jb2xvcj0iI2NjYyIgb2Zmc2V0PSI1MCUiIC8+CiAgICAgIDxzdG9wIHN0b3AtY29sb3I9IiNjY2MiIG9mZnNldD0iNzAlIiAvPgogICAgPC9saW5lYXJHcmFkaWVudD4KICA8L2RlZnM+CiAgPHJlY3Qgd2lkdGg9IjQwMCIgaGVpZ2h0PSIzMDAiIGZpbGw9IiNjY2MiIC8+CiAgPHJlY3QgaWQ9InIiIHdpZHRoPSI0MDAiIGhlaWdodD0iMzAwIiBmaWxsPSJ1cmwoI2cpIiAvPgogIDxhbmltYXRlIHhsaW5rOmhyZWY9IiNyIiBhdHRyaWJ1dGVOYW1lPSJ4IiBmcm9tPSItNDAwIiB0bz0iNDAwIiBkdXI9IjFzIiByZXBlYXRDb3VudD0iaW5kZWZpbml0ZSIgIC8+Cjwvc3ZnPg==);"
          />
          <noscript />
        </span>
      </div>
    </div>
    <div
      class="chakra-card__body css-1278jd6"
    >
      <div
        class="chakra-stack css-1k4nord"
        style="margin: 0px;"
      >
        <h2
          class="chakra-heading css-1n2yjyv"
        >
          Plants make the atmosphere of the house come alive
        </h2>
        <p
          class="chakra-text css-aaaj5s"
        >
          Houseplants are good for your health — and not just for their visual beauty. Why? They essentially do the opposite of what we do when we breathe: release oxygen and absorb carbon dioxide. This not only freshens up the air, but also eliminates harmful toxins. Extensive research by NASA has revealed that houseplants can remove up to 87 per cent of air toxin in 24 hours. Studies have also proven that indoor plants improve concentration and productivity (by up to 15 percent!), reduce stress levels and boost your mood — making them perfect for not just your home but your work space, too. At work, place plants, especially those with broad leaves, on your desk; they will help regulate humidity and increase levels of positivity — seeing greenery and nature help us feel more relaxed and calm, which in turn benefits your every day mood. Indoor plants serve a practical and aesthetic purpose, and will enhance your life. How to Create your Personal Plant Sanctuary. An indoor garden can be your refuge from the outside world, and for many people it is a source of great joy. Whether you live in a small apartment, or a large house, by introducing certain plants into your home, you will start to notice improvements to your health, and overall happiness. As well as enhancing your mood and creating a living space that is soothing to be in, plants can also help with loneliness and depression: caring for a living thing gives us a purpose and is rewarding — especially when you see that living thing bloom and thrive. choose the right plants for the an optimal nights sleep. Although plants release oxygen during the day, it is worth remembering that, at night, when photosynthesis stops, most plants switch things up and release carbon dioxide. However plants such as orchids, succulents, snake plants and bromeliads do the opposite and emit oxygen, making them perfect plants for the bedroom (and getting better sleep). Beware of too much sun. Most indoor plants don’t like direct midday sun, so please be wary of this when placing plants in your home. There are very obvious warning signs to look out for, such as leaf burn, spotting or sudden leaf-fall. Most plants can be easily rescued, so don’t panic! Often it is a case of not watering them too much, or letting the soil dry out a bit if it is feeling too soggy. Check if your plant is sitting by a cold draft as this can case the leaves to curl and eventually drop off. Organic fertilizers are a great way to revive your plants.
        </p>
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
